### DISCUSSION TITLE
The only classnames substitute that does not have side effects 👏

### DISCUSSION THREAD

**Discussion by seriema** [xobotyi/cnbuilder#377] [Show and tell]
Just wanted to say thank you. You deserve more stars. ⭐⭐⭐⭐⭐

I have tried replacing classnames with lukeed/clsx and jorgebucaran/classcat. Rollup fails to determine side effects in both of them. Your package is correctly marked with [`sideEffects: false` in package.json](https://webpack.js.org/guides/tree-shaking/) which is a huge boon IMHO.

Just wanted to say thank you and keep it up! I saw you have 229 commits to master since the last release. Looking forward to what that is. 😄

**Comment by xobotyi** [comment]
Ah, it's just dependabot spamming a bit😅
As for now I can't see any stuff to improve tbh, and no more releases expected (at least for me, but who knows).

The only thing to do is to promote the package, but I've already gave up, don't know how to do it, spamming at reddit gave almost nothing😔

**Comment by seriema** [comment]
I only found it on reddit after already finding it on npm and somewhere else.

Write a blog post about it?

**Comment by seriema** [comment]
Maybe better. Do some PR's to popular repos that still use classnames? Replace them with cnbuilder and explain the benefits in the PR to the maintainer.

Our rollup still detects classnames as a side-effects file because it's coming through the components in [RMWC](https://rmwc.io/). If I could wish for a PR, that'd be the one. You can see it's referencing classnames like this:
https://github.com/jamesmfriedman/rmwc/search?q=classnames

**Comment by xobotyi** [comment]
Ive tried, most of maintainers be like "Eeeeh, dont wanna change thing that working. What? only 2 times faster? pff dont even bother"
TBH the point of "dont touch if its working" is a good one =)

**Comment by xobotyi** [comment]
But thanks for the share - i'll try to make pr to rmwc some time later.

**Comment by seriema** [comment]
Speed wasn’t the main thing for me. Too many competing on that already. It was the packaging that made rollup understand it can tree shake that was worth it. Not sure why all classname alternatives try to be so drop-in replacements that they need to be default exports in commonjs. 🤷🏽‍♂️

**Comment by xobotyi** [comment]
Well default cjs default exports are mystery for mee too. Everyone anyway building with bundlers and somewhy developers still trying to satisfy these .5% masochists that uses cjs or worse umd🤦‍♀️

**Comment by xobotyi** [comment]
@seriema by the way AFAIK there is plugin for rollup that gives the ability to replace modules build time. And since cnb can replace any of classnames, clsx or classcat without code change - you can use it easily

**Comment by seriema** [comment]
@xobotyi I gave it a few attempts to find what you meant and configure rollup to replace classname with cnbuilder but didn't get it to work. Do you have an example?

**Comment by xobotyi** [comment]
@seriema I remember right - this one: https://www.npmjs.com/package/@rollup/plugin-alias
